### PR TITLE
Close dropdowns

### LINKED ISSUE #7477: When a dropdown is open Escape should close the dropdown and Enter should still add a new node.
If dropdown widget is open can use cursor keys.

Escape should close the dropdown.
Enter should still add a new now and close dropdown.

### LINKED ISSUE #7562: Dropdowns not disappearing on selecting another node and are still in selection stage
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

When I click the dropdown on a widget it stays active even when I select another component, node. It only disappears when I click on the empty canvas.

https://github.com/enso-org/enso/assets/108271897/2b3e710f-d48f-4265-ac48-d5e910916472

### Expected behaviour

It should disappear on clicking whatever outside the selection of the dropdown.

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Logs

_No response_

### Enso Version

2023.8.11 nightly

### Browser or standalone distribution

Standalone distribution (local project)

### Browser Version or standalone distribution

standalone

### Operating System

Windows

### Operating System Version

Win11pro 22H2 22621.1848

### Hardware you are using

12th Gen Intel(R) Core(TM) i9-12900HK / RTX3060 Laptop / Nvidia Drivers 531.68

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Fixes #7562. Close a dropdown when:
- A click outside the dropdown occurs
- `Esc` is pressed
- Any other `Interaction` is started (i.e. using a shortcut)

### Important Notes

- Simplifies `Interaction` API and uses it for closing/canceling CB as well as dropdowns.
- Adjusted some event handlers so that handled clicks don't also register as GraphEditor background clicks.
- Introduces a CSS approach to prevent unwanted text-selections; we were doing it with JS until my previous PR, and the JS solution was breaking things.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor.vue [hunk 1]
```diff
@@ -669,11 +617,7 @@ function handleEdgeDrop(source: AstId, position: Vec2) {
       @zoomIn="graphNavigator.scale *= 1.1"
       @zoomOut="graphNavigator.scale *= 0.9"
     />
-    <PlusButton
-      @click.stop="interaction.setCurrent(creatingNodeFromButton)"
-      @pointerdown.stop
-      @pointerup.stop
-    />
+    <PlusButton @pointerdown.stop="startCreatingNodeFromButton()" @click.stop @pointerup.stop />
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor.vue]: It should be click, not pointerdown. All basic "button clicks" should be now click, not pointerdowns.

File: app/gui2/src/components/GraphEditor.vue [hunk 2]
```diff
@@ -634,7 +582,7 @@ function handleEdgeDrop(source: AstId, position: Vec2) {
     :style="groupColors"
     v-on.="graphNavigator.events"
     v-on..="nodeSelection.events"
-    @click="handleClick"
+    @pointerdown="handleClick"
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor.vue]: Why this change? I deliberately changed it to click once.

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 3]
```diff
@@ -119,6 +125,19 @@ const innerWidgetInput = computed(() => {
...
+  },
+  click: (e: PointerEvent) => {
+    if (targetIsOutside(e, widgetRoot)) showDropdownWidget.value = false
+    return false
+  },
+})
+
+const handleClick = defineKeybinds(`dropdown-${random.uint53()}`, {
+  toggleDropdownWidget: ['PointerMain'],
+}).handler({ toggleDropdownWidget })
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: All `defineKebinds` should be in bindings.ts, and those bindings might be read from file/settings in the future. Users will rather not set different binding for every drop-down widget.

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 4]
```diff
@@ -172,7 +191,13 @@ export const widgetDefinition = defineWidget(WidgetInput.isAstOrPlaceholder, {
 
 <template>
   <!-- See comment in GraphNode next to dragPointer definition about stopping pointerdown and pointerup -->
-  <div class="WidgetSelection" @pointerdown.stop @pointerup.stop @click.stop="toggleDropdownWidget">
+  <div
+    ref="widgetRoot"
+    class="WidgetSelection"
+    @pointerdown.stop="handleClick"
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: The click should be the default way of handling.

**Review by farmaazon — APPROVED** [review]

File: app/gui2/src/components/GraphEditor.vue (hunk 2 above)
Comment (kazcw) [reply app/gui2/src/components/GraphEditor.vue]: I was preventing some user actions triggering multiple handlers, but we can do that some other way. Fixing it isn't needed for this PR and it doesn't currently cause any observed problems.

**Pull request merged**